tsinfer/inference.py

```diff
@@ -1907,6 +1909,149 @@ def get_mutations(self, node_id):
         return self.mutations[node_id]
 
 
+def has_single_edge_over_grand_root(ts):
+    # Internal function to check if this is a "raw" inferred tree sequence.
+    if ts.num_edges < 2:
+        # must have edge to grand root and above grand root
+        return False
+    last_edge = ts.edge(-1)
+    if ts.edge(-2).parent == last_edge.parent:
+        return False  # other edges point to the oldest node => not a virtual-like root
+    if last_edge.left != 0 or last_edge.right != ts.sequence_length:
+        return False  # edge does not span the entire genome
+    return True
+
+
+def post_process(
+    ts,
+    *,
+    split_mrca=None,
+):
+    """
+    Post-process a tsinferred tree sequence into a more conventional form. This is
+    the function run by default on the final tree sequence output by
+    :func:`match_samples`. It involves removing the oldest node if it sits above a
+    grand MRCA ancestor (this "virtual-root-like node" is used to enable LS matching)
+    then splitting up the grand MRCA (if split_mrca is ``True``), and finally
+    reordering the sample nodes such that they are the first nodes listed in the
+    node table,  removing tree nodes and edges that are not on a path between the
+    root and any of the samples (by applying the :meth:`tskit.TreeSequence.simplify`
+    method with ``keep_unary`` set to True but ``filter_sites``, ``filter_populations``
+    and ``filter_individuals`` set to False).
+
+    :param bool split_mrca: If the oldest node is only a parent to a single
+        "grand MRCA" of all the samples, after removing the oldest node we can
+        split the grand MRCA into separate nodes whenever its children change.
+        The rationale is that a grand MRCA consisting of a single ancestral haplotype
+        with all inference sites in the ancestral state is unlikely to represent a
+        single ancestor in the past. If this parameter is ``False``, any such grand
+        MRCA is *not* split (warning: this may cause problems when further dating the
+        inferred tree sequence). Default: ``True``.
+    """
+    if split_mrca is None:
+        split_mrca = True
+
+    tables = ts.dump_tables()
+
+    if has_single_edge_over_grand_root(ts):
+
+        logger.info("Removing the oldest edge to detach the virtual-root-like ancestor")
+        last_edge = ts.edge(
+            -1
+        )  # Edge with oldest parent must be last in the edge table
```

Comment:
I would either remove this comment or put it on the line above so we don't get this line-splitting from Black.